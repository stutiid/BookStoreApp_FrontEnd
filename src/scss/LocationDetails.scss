.location-main {
  width: 100%;
  display: flex;
  font-size: 20px;
  align-items: center;
  justify-content: flex-start;
  transform: translate(-50%, -50%);
}
.location-card {
  text-align: left;
  height: 300px;
  width: 500px;
  margin-top: 100px;
}
