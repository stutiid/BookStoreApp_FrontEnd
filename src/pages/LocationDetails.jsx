import { Card, CardContent } from "@mui/material";
import { Component } from "react";
import "../scss/LocationDetails.scss";

/**
 * @description:- class represents a particular location details which is to be shown on the map
 */
export default class LocationDetails extends Component {
  constructor(props) {
    super(props);
    this.state = {
      icon: this.props.icon,
      lat: this.props.lat,
      long: this.props.long,
      text: this.props.text,
      description: this.props.description,
      state: this.props.state,
      population: this.props.population,
      display: "none",
    };
  }

  /**
   * @description:- to handle the mouse hover event and then setting the display state to block so to show the other details about
   * the location on the map.
   */
  handleMouseEntry = () => {
    this.setState({
      display: "block",
    });
  };

  /**
   * @description:- to handle the mouse leaving event and then setting the display state to none so to hide the other details about
   * the location
   */
  handleMouseLeave = () => {
    this.setState({
      display: "none",
    });
  };

  /**
   * @description :- details about the location which are to be renderd on the map
   */
  render() {
    return (
      <div
        className="location-main"
        onMouseEnter={this.handleMouseEntry}
        onMouseLeave={this.handleMouseLeave}
      >
        {this.state.icon}
        {this.state.text}
        <div
          style={{
            display: this.state.display,
          }}
        >
          <Card
            className="location-card"
            style={{ backgroundColor: "lightyellow" }}
          >
            <CardContent>
              <b>
                <u>Description:-</u>
              </b>{" "}
              {this.state.description}
              <br />
              <b>
                <u>State:-</u>
              </b>{" "}
              {this.state.state}
              <br />
              <b>
                <u>Population:-</u>
              </b>{" "}
              {this.state.population}
            </CardContent>
          </Card>
        </div>
      </div>
    );
  }
}
