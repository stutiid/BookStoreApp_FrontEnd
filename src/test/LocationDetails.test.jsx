import "./setupTest";
import LocationDetails from "../pages/LocationDetails";
import { shallow } from "enzyme";

const data = {
  id: 1,
  name: "Banglore",
  latitude: "12.9699",
  longitude: "77.5980",
  description:
    "Bengaluru (also called Bangalore) is the capital of India's southern Karnataka state.",
  state: "Karnataka",
  population: 13707000,
};

describe("tests", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<LocationDetails data={data} />);
  });

  afterEach(() => {
    wrapper.unmount();
  });

  it("should render initial layout", () => {
    expect(wrapper.getElements()).toMatchSnapshot();
  });

  it("has a div containing icon and name of location", () => {
    expect(wrapper.find(".location-main")).toBeTruthy();
  });

  it("has a card to display other details about the location", () => {
    expect(wrapper.find(".location-card")).toBeTruthy();
  });
});
