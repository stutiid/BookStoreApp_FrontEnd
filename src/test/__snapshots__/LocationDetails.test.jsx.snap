// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests should render initial layout 1`] = `
Array [
  <div
    className="location-main"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
  >
    <div
      style={
        Object {
          "display": "none",
        }
      }
    >
      <ForwardRef(Card)
        className="location-card"
        style={
          Object {
            "backgroundColor": "lightyellow",
          }
        }
      >
        <ForwardRef(CardContent)>
          <b>
            <u>
              Description:-
            </u>
          </b>
           
          <br />
          <b>
            <u>
              State:-
            </u>
          </b>
           
          <br />
          <b>
            <u>
              Population:-
            </u>
          </b>
           
        </ForwardRef(CardContent)>
      </ForwardRef(Card)>
    </div>
  </div>,
]
`;
